javav2/example_code/ses/src/test/java/SESTest.java

```diff
@@ -82,61 +80,56 @@ public void whenInitializingAWSService_thenNotNull() {
     @Test
     @Order(2)
     public void SendMessage() {
-
-        try {
-            SendMessage.send(client, sender,recipient, subject,bodyText,bodyHTML);
-            System.out.println("Test 2 passed");
-
-        } catch (IOException | MessagingException e) {
-            e.getStackTrace();
-        }
+        assertDoesNotThrow(() -> SendMessage.send(client, sender,recipient, subject,bodyText,bodyHTML));
+        System.out.println("Test 2 passed");
     }
 
     @Test
     @Order(3)
     public void SendMessageV2() {
-
-      SendEmail.send(sesv2Client, sender, recipient, subject, bodyHTML);
-      System.out.println("Test 3 passed");
+        assertDoesNotThrow(() -> SendEmail.send(sesv2Client, sender, recipient, subject, bodyHTML));
+        System.out.println("Test 3 passed");
     }
 
     @Test
     @Order(4)
     public void SendMessageAttachment() {
-
-        try {
-            SendMessageAttachment.sendemailAttachment(client, sender,recipient, subject,bodyText,bodyHTML,fileLocation );
-            System.out.println("Test 4 passed");
-
-        } catch (IOException | MessagingException e) {
-            e.getStackTrace();
-        }
+        assertDoesNotThrow(() -> SendMessageAttachment.sendemailAttachment(client, sender, recipient, subject, bodyText, bodyHTML, fileLocation ));
+        System.out.println("Test 4 passed");
     }
 
     @Test
     @Order(5)
     public void SendMessageAttachmentV2() {
-
-        try {
-            com.example.sesv2.SendMessageAttachment.sendEmailAttachment(sesv2Client, sender, recipient, subject, bodyHTML, fileLocation );
-            System.out.println("Test 5 passed");
-
-        } catch (IOException | MessagingException e) {
-            e.getStackTrace();
-        }
+        assertDoesNotThrow(() -> com.example.sesv2.SendMessageAttachment.sendEmailAttachment(sesv2Client, sender, recipient, subject, bodyHTML, fileLocation));
+        System.out.println("Test 5 passed");
     }
 
     @Test
     @Order(6)
-    public void ListIdentities() {
-        ListIdentities.listSESIdentities(client);
+    public void SendMessageTemplateV2() {
+        assertDoesNotThrow(() -> SendEmailTemplate.send(sesv2Client, sender, recipient, templateName));
         System.out.println("Test 6 passed");
     }
 
     @Test
     @Order(7)
+    public void ListIdentities() {
+        assertDoesNotThrow(() -> ListIdentities.listSESIdentities(client));
+        System.out.println("Test 6 passed");
+    }
+
+    @Test
+    @Order(8)
     public void ListEmailIdentities() {
-        ListEmailIdentities.listSESIdentities(sesv2Client);
+        assertDoesNotThrow(() -> ListEmailIdentities.listSESIdentities(sesv2Client));
+        System.out.println("Test 4 passed");
+    }
+
+    @Test
+    @Order(9)
+    public void ListEmailTemplates() {

```

Comment:
Something to think about for this type of test, is that since a form of output from the method might be thought of as "A List of Email Templates" it would be nice to have a `List<EmailTemplates>` or something like that to test against. Then, with known data, we could check that the size of the list was 4 and that index 1 of the list has X data.

We can't do that here, because the method doesn't have a return type and instead prints out the list of items. If we think of this in MVC terms, that's mixing Model and View layers. It would be more reusable and testable if `listAllTemplates` gave us a list of templates, and then in the main method of ListTemplates, we could have 

```
templates = listALlTemplates(sesv2Client);
templates.forEach(template ->
                    System.out.println("Template name: " + template.templateName()));
```
Just some thoughts.

Reply:
Well consider this moving forward.